xivo_bus/mixins.py

```diff
@@ -0,0 +1,421 @@
+# Copyright 2021 The Wazo Authors  (see the AUTHORS file)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import os
+import sys
+
+from threading import Thread, Lock, Semaphore
+from inspect import isclass
+from six import raise_from, iteritems
+from contextlib import contextmanager
+from collections import namedtuple, defaultdict
+from kombu import Queue, Exchange, Connection, Producer, binding as Binding
+from kombu.mixins import ConsumerMixin as KombuConsumer
+
+from .publishing_queue import PublishingQueue
+from .middlewares import Middleware, MiddlewareError
+
+
+class InitMixin(object):
+    def __init__(
+        self,
+        username='guest',
+        password='guest',
+        host='localhost',
+        port=5672,
+        exchange_name=None,
+        exchange_type=None,
+        middlewares=None,
+        **kwargs
+    ):
+        super(InitMixin, self).__init__(
+            username=username,
+            password=password,
+            host=host,
+            port=port,
+            exchange_name=exchange_name,
+            exchange_type=exchange_type,
+            middlewares=middlewares,
+            **kwargs
+        )
+
+
+class ThreadableMixin(object):
+    def __init__(self, **kwargs):
+        super(ThreadableMixin, self).__init__(**kwargs)
+        self.should_stop = False
+
+    @property
+    def is_running(self):
+        try:
+            return any({thread.is_alive() for thread, _, _ in self.__threads})
+        except AttributeError:
+            return False
+
+    @property
+    def __threads(self):
+        try:
+            return self.__t
+        except AttributeError:
+            self.__t = list()
+            return self.__t
```

Comment:
Since Consumer and Publisher can no more be in the same object, no need to support multiple threads for this mixin, I think?